managed-identity-storage-blob/src/main/java/com/microsoft/azure/MainController.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+package com.microsoft.azure;
+
+import com.azure.identity.ManagedIdentityCredential;
+import com.azure.identity.ManagedIdentityCredentialBuilder;
+import com.azure.storage.blob.BlobContainerClient;
+import com.azure.storage.blob.BlobServiceClient;
+import com.azure.storage.blob.BlobServiceClientBuilder;
+import com.azure.storage.blob.specialized.BlockBlobClient;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.web.bind.annotation.*;
+
+import javax.annotation.PostConstruct;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.util.Locale;
+
+@RestController
+public class MainController {
+    private BlobContainerClient blobContainerClient;
+
+    @Value("${azure.storage.account:local}")
+    private String accountName;
+
+    @Value("${azure.storage.container:local}")
+    private String containerName;
+
+    @PostConstruct
+    private void setupBlobClient() {
+        ManagedIdentityCredential managedIdentityCredential = new ManagedIdentityCredentialBuilder()
+                .maxRetry(1)
+                .retryTimeout(duration -> Duration.ofMinutes(1))
+                .build();
+
+        String endpoint = String.format(Locale.ROOT, "https://%s.blob.core.windows.net", accountName);

```

Comment:
Can we not hard-code the endpoint here?
Once we support national clouds like mooncake, blackforest, we will have to change this sample, right?

Reply:
Yes, I' afraid so. This behavior is consistent with azure-spring-boot starter, which also need to be updated after onboarding to national clouds. Will sync with Xiaolu about that

Reply:
https://github.com/microsoft/azure-spring-boot/issues/898 we will fix this in this sprint